doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,167 +866,137 @@ Module system
-       with body :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} <: {+<: @module_type} := @module_expression
-
-       Defines a functor with parameters given by module_bindings (possibly none) with body :token:`module_expression`.
-       The body is checked against each :n:`@module_type__i`.
-
-    .. cmdv:: Module @ident {* @module_binding} := {+<+ @module_expression}
-
-       is equivalent to an interactive module where each :token:`module_expression` is included.
-
-.. cmd:: Module Type @ident
+.. note::
 
-   This command is used to start an interactive module type :token:`ident`.
+  #. Interactive modules and module types can be nested.
+  #. Interactive modules and module types can't be defined inside of :ref:`sections<section-mechanism>`.
+     Sections can be defined inside of interactive modules and module types.
+  #. Hints and notations (:cmd:`Hint` and :cmd:`Notation` commands) can also appear inside interactive
+     modules and module types. Note that with a module definition like:
 
-   .. cmdv:: Module Type @ident {* @module_binding}
+  ::
 
-      Starts an interactive functor type with parameters given by :token:`module_bindings`.
+     Module N : SIG := M.
 
+  or::
 
-Reserved commands inside an interactive module type:
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Module N : SIG. … End N.
 
-.. cmd:: Include @module
+  hints and the like valid for ``N`` are not those defined in ``M``
+  (or the module body) but the ones defined in ``SIG``.
 
-   Same as ``Include`` inside a module.
+.. _terminating_module:
 
-.. cmd:: Include {+<+ @module}
+Terminating an interactive module or module type definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   This is a shortcut for the command :n:`Include @module` for each :token:`module`.
+.. todo this size is overkill for such a short section
 
-.. cmd:: @assumption_token Inline @assums
-   :name: Inline
+.. todo: description below applies equally to modules and module types?
 
-   The instance of this assumption will be automatically expanded at functor application, except when
-   this functor application is prefixed by a ``!`` annotation.
+Interactive modules are terminated with the :cmd:`End` command, which
+is also used to terminate :ref:`Sections<section-mechanism>`.
+:n:`End @ident` closes the interactive module or module type :token:`ident`.
+If the module type was given, the command verifies that the content of the module
+matches the module type.  If the module is basic (not a
+functor), its components (constants, inductive types, submodules etc.)
+are now available through the dot notation.
 
-.. cmd:: End @ident
+.. exn:: No such label @ident.
+    :undocumented:
 
-   This command closes the interactive module type :token:`ident`.
+.. exn:: Signature components for label @ident do not match.
+    :undocumented:
```

Comment:
Another existing undocumented error is "The field @ident is missing in @qualid".